File: LeetCode/1512.Number_Of_Good_Pairs/main.cpp
#include "solution.h"

// Unit Test

int main(){
    Solution s;
    vector<int> v1 {1,2,3,1,1,3};
    vector<int> v2 {1,1,1,1};
    vector<int> v3 {1,2,3};
    if (s.numIdenticalPairs(v1) == 4) {cout << "Pass" << endl; } else {cout << "Failed 1" << endl; }
    if (s.numIdenticalPairs(v2) == 6) {cout << "Pass" << endl; } else {cout << "Failed 2" << endl; }
    if (s.numIdenticalPairs(v3) == 0) {cout << "Pass" << endl; } else {cout << "Failed 3" << endl; }
    
    return 0;
}

File: LeetCode/1512.Number_Of_Good_Pairs/main.py
#!/usr/bin/python

from solution import Solution

# Unit Test
nums1 = [1,2,3,1,1,3]
nums2 = [1,1,1,1]
nums3 = [1,2,3]
s = Solution()
if (s.numIdenticalPairs(nums1) == 4):
    print("Pass")
else:
    print("Test 1 Failed")
if (s.numIdenticalPairs(nums2) == 6):
    print("Pass")
else:
    print("Test 2 Failed")
if (s.numIdenticalPairs(nums3) == 0):
    print("Pass")
else:
    print("Test 3 Failed")